ImGuiFileDialog.cpp

```diff
@@ -75,300 +103,330 @@ ImGuiFileDialog::~ImGuiFileDialog()
-	{
-		if (itPathDecomp != m_CurrentPath_Decomposition.begin())
-			ImGui::SameLine(); 
-		if (ImGui::Button((*itPathDecomp).c_str()))
-		{
-			ComposeNewPath(itPathDecomp);
-			pathClick = true;
-			break;
-		}
-	}
-
-	ImVec2 size = ImGui::GetContentRegionMax() - ImVec2(0.0f, 120.0f);
-
-	ImGui::BeginChild("##FileDialog_FileList", size);
-
-	for (std::vector<FileInfoStruct>::iterator it = m_FileList.begin(); it != m_FileList.end(); ++it)
-	{
-		FileInfoStruct infos = *it;
-		
-		bool show = true;
-
-		std::string str;
-		if (infos.type == 'd') str = "[Dir] " + infos.fileName;
-		if (infos.type == 'l') str = "[Link] " + infos.fileName;
-		if (infos.type == 'f') str = "[File] " + infos.fileName;
-		if (infos.type == 'f' && m_CurrentFilterExt.size() > 0 && infos.ext != m_CurrentFilterExt)
-		{
-			show = false;
-		}
-		if (show == true)
-		{
-			if (ImGui::Selectable(str.c_str(), (infos.fileName == m_SelectedFileName)))
-			{
-				if (infos.type == 'd')
-				{
-					if (infos.fileName == "..")
-					{
-						if (m_CurrentPath_Decomposition.size() > 1)
-						{
-							std::vector<std::string>::iterator itPathDecomp = m_CurrentPath_Decomposition.end() - 2;
-							ComposeNewPath(itPathDecomp);
-						}
-					}
-					else
-					{
-						m_CurrentPath += "\\" + infos.fileName;
-					}
-					pathClick = true;
-				}
-				else
-				{
-					m_SelectedFileName = infos.fileName;
-					ResetBuffer(FileNameBuffer);
-					AppendToBuffer(FileNameBuffer, MAX_FILE_DIALOG_NAME_BUFFER, m_SelectedFileName);
-				}
-				break;
-			}
-		}
-	}
-
-	// changement de repertoire
-	if (pathClick == true)
-	{
-		ScanDir(m_CurrentPath);
-		m_CurrentPath_Decomposition = splitStringVector(m_CurrentPath, '\\');
-		if (m_CurrentPath_Decomposition.size() == 2)
-			if (m_CurrentPath_Decomposition[1] == "")
-				m_CurrentPath_Decomposition.erase(m_CurrentPath_Decomposition.end() - 1);
-	}
-
-	ImGui::EndChild();
-
-	ImGui::Text("File Name : ");
-
-	ImGui::SameLine();
-
-	float width = ImGui::GetContentRegionAvailWidth();
-	if (vFilters != 0) width -= 120.0f;
-	ImGui::PushItemWidth(width);
-	ImGui::InputText("##FileName", FileNameBuffer, MAX_FILE_DIALOG_NAME_BUFFER);
-	ImGui::PopItemWidth();
-
-	if (vFilters != 0)
-	{
-		ImGui::SameLine();
-
-		ImGui::PushItemWidth(100.0f);
-		bool comboClick = ImGui::Combo("##Filters", &FilterIndex, vFilters) || m_CurrentFilterExt.size() == 0;
-		ImGui::PopItemWidth();
-		if (comboClick == true)
-		{
-			int itemIdx = 0;
-			const char* p = vFilters;
-			while (*p)
-			{
-				if (FilterIndex == itemIdx)
-				{
-					m_CurrentFilterExt = std::string(p);
-					break;
-				}
-				p += strlen(p) + 1;
-				itemIdx++;
-			}
-		}
-	}
-	
-	if (ImGui::Button("Cancel"))
-	{
-		IsOk = false;
-		res = true;
-	}
-
-	ImGui::SameLine();
-
-	if (ImGui::Button("Ok"))
-	{
-		IsOk = true;
-		res = true;
-	}
-
-	ImGui::End();
-
-	if (res == true)
-	{
-		m_FileList.clear();
-	}
-
-	return res;
+    bool res = false;
+
+    IsOk = false;
+
+    ImGui::OpenPopup(vName);
```

Comment:
I wanted to have the dialog modal. It would be good to have both ways of doing it. Note that when it pops up my dialog is too small, so my code here still needs some fixing.